org-code-javabuilder/media/src/main/java/org/code/media/Image.java

```diff
@@ -151,6 +164,17 @@ public BufferedImage getBufferedImage() {
     return bufferedImageResult;
   }
 
+  /**
+   * Load the given image from a URL and return an Image object
+   *
+   * @param url to load the file from
+   * @return Image object
+   * @throws FileNotFoundException if the file is not found at the URL
+   */
+  public static Image fromUrl(URL url) throws FileNotFoundException {
```

Comment:
Is there a way for us to protect this function better? I am slightly concerned about students being able to get images from arbitrary urls, although with import blocking I think this concern may go away (since they won't be able to access `URL`).

Reply:
Yeah agreed - I added this static method in an effort to protect it a little (as opposed to adding an `Image` constructor that takes a URL) but it is still fully public and accessible. I think for now I can add a different static method `fromMediaFile` or something that will internally fetch the uploaded image.

Looking at this though, I'm also thinking in the long run we could refactor how we access media content in general, especially as we consider different ways of storing Javabuilder content as per our infra discussion - maybe some `JavabuilderContentManager` that manages all the content Javabuilder needs to run (media, source code, etc).